ui/src/timeMachine/reducers/index.test.ts

```diff
@@ -189,4 +191,37 @@ describe('the Time Machine reducer', () => {
       expect(activeTimeMachine.activeQueryIndex).toBe(originalActiveQueryIndex)
     })
   })
+
+  // Fix for: https://github.com/influxdata/influxdb/issues/17364
+  describe('editing a table view', () => {
+    it('does not overwrite internal TableViewProperites when files is an empty array', () => {
+      const initial = initialState()
+
+      const store = createStore(timeMachinesReducer, initial)
+      store.dispatch({type: 'SET_VIEW_TYPE', payload: {type: 'table'}})
+
+      const midState = store.getState()
+      const tableViewProperties = midState.timeMachines[
+        midState.activeTimeMachineID
+      ].view.properties as TableViewProperties
+      tableViewProperties.fieldOptions = [
+        {internalName: '_foo', displayName: 'foo'},
+      ]
+
+      store.dispatch({
+        type: 'SET_QUERY_RESULTS',
+        payload: {
+          status: RemoteDataState.Done,
+          files: [],
+          fetchDuration: 234,
+        },
+      })
+
+      const endState = store.getState()
+      expect(
+        (endState.timeMachines[endState.activeTimeMachineID].view
+          .properties as TableViewProperties).fieldOptions[0].displayName
+      ).toBe(tableViewProperties.fieldOptions[0].displayName)
```

Comment:
```suggestion
     const tvp = endState.timeMachines[endState.activeTimeMachineID].view
          .properties as TableViewProperties
          
     const actual = tvp.fieldOptions[0].displayName
     const expected = tableViewProperties.fieldOptions[0].displayName
      
      expect(actual).toBe(expected)
```
or some such cleaner thing to look at